### PR TITLE
[risk=low][RW-7573] add a partitionColumn field to the actionAudit config

### PR DESCRIPTION
**Pull request by jmthibault79**
Preprod and Staging are configured differently from other envs, so we need to account for this in the app config.

Tested by running user audit locally

TODO: update the config in Terraform ([RW-7601](https://precisionmedicineinitiative.atlassian.net/browse/RW-7601))

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/config/config_preprod.json [hunk 1]
```diff
@@ -111,7 +111,8 @@
   "actionAudit": {
     "logName": "workbench-action-audit-preprod",
     "bigQueryDataset": "workbench_action_audit_preprod",
-    "bigQueryTable": "workbench_action_audit_preprod"
+    "bigQueryTable": "workbench_action_audit_preprod",
+    "partitionColumn": "timestamp"
```
Comment (jmthibault79) [inline api/config/config_preprod.json:115]: this was the cause of the bug

File: api/config/config_staging.json [hunk 2]
```diff
@@ -110,8 +110,9 @@
   },
   "actionAudit": {
     "logName": "workbench-action-audit-staging",
-    "bigQueryDataset": "workbench_action_audit_staging",
-    "bigQueryTable": "workbench_action_audit_staging"
+    "bigQueryDataset": "workspace_action_audit_staging",
```
Comment (jmthibault79) [inline api/config/config_staging.json:113]: somehow this was configured as work**space** not work**bench**.  I suspect auditing doesn't work in Staging either.

File: api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java [hunk 3]
```diff
@@ -76,10 +75,6 @@ public WorkspaceAuditLogQueryResponse queryEventsForWorkspace(
 
     final TableResult tableResult = bigQueryService.executeQuery(queryJobConfiguration);
     final List<AuditLogEntry> logEntries = auditLogEntryMapper.tableResultToLogEntries(tableResult);
-    final String queryHeader =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java:79]: not used

File: api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java [hunk 4]
```diff
@@ -120,10 +115,6 @@ public UserAuditLogQueryResponse queryEventsForUser(
     final TableResult tableResult = bigQueryService.executeQuery(queryJobConfiguration);
 
     final List<AuditLogEntry> logEntries = auditLogEntryMapper.tableResultToLogEntries(tableResult);
-    final String queryHeader =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java:123]: not used

File: api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java [hunk 5]
```diff
@@ -39,8 +39,7 @@
           + "WHERE %s\n"
           + "  AND @after <= TIMESTAMP_MILLIS(CAST(jsonPayload.timestamp AS INT64))\n"
           + "  AND TIMESTAMP_MILLIS(CAST(jsonPayload.timestamp AS INT64)) < @before\n"
-          + "  AND @after_partition_time <= _PARTITIONTIME\n"
-          + "  AND _PARTITIONTIME < @before_partition_time\n"
+          + "  AND @after_partition_time <= %s AND %<s < @before_partition_time\n"
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java:42]: TIL `%<s` means repeat the previous instance of `%s`

Comment (dolbeew) [reply api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java:42]: Good to know!

**Review by dolbeew — APPROVED** [review]

**Pull request merged**